#! /usr/bin/env python
#Autogenerated V4L Dynamic Control

PACKAGE='mrpt_rawlog'
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("debug", bool_t, 0, "Debug", False)
enum_motion_noise_type = gen.enum([gen.const("gaussian", int_t, 0, "gaussian"), gen.const("Thrun", int_t, 1, "Thrun")], "motion type")
gen.add("motion_noise_type", int_t, 0, "motion noise", 0, 0, 1, edit_method=enum_motion_noise_type)
gen.add("motion_gaussian_alpha_1", double_t, 0, "motion model: alpha_1", 0.034, 0, 10)
gen.add("motion_gaussian_alpha_2", double_t, 0, "motion model: alpha_2", 0.057, 0, 10)
gen.add("motion_gaussian_alpha_3", double_t, 0, "motion model: alpha_3", 0.014, 0, 10)
gen.add("motion_gaussian_alpha_4", double_t, 0, "motion model: alpha_4", 0.097, 0, 10)
gen.add("motion_gaussian_alpha_xy", double_t, 0, "motion model: alpha_xy", 0.005, 0, 10)
gen.add("motion_gaussian_alpha_phi", double_t, 0, "motion model: alpha_phi", 0.05, 0, 10)

gen.add("bearing_range_std_range", double_t, 0, "sensor model: standard deviation of range", 0.01, 0, 10)
gen.add("bearing_range_std_yaw", double_t, 0, "sensor model: standard deviation of yaw", 0.01, 0, 10)
gen.add("bearing_range_std_pitch", double_t, 0, "sensor model: standard deviation of pitchal", 0.00, 0, 10)

gen.add("sensor_frame_sync_threshold", double_t, 0, "maximal time difference [sec] to get observations into the same sensor frame", 0.01, 0, 10)

exit(gen.generate(PACKAGE, "mrpt_rawlog", "RawLogRecord"))
